trackintel/preprocessing/staypoints.py

```diff
@@ -119,7 +119,7 @@ def generate_locations(
             # fill first nan with 0 and create the cumulative sum
             loc_id_offset = loc_id_offset.fillna(0).cumsum()
 
-            ret_stps_non_noise_labels["location_id"] = ret_stps["location_id"] + loc_id_offset
+            ret_stps_non_noise_labels["location_id"] = ret_stps_non_noise_labels["location_id"] + loc_id_offset
```

Comment:
This seems to be a clear bug and pretty much careless! sorry for introducing it! 
I am just trying to understand if we need a new test for this or a minor tweak in one of the other tests checks for the stability of indices?

Reply:
Apart from the need for the new tests, the change is fine. I will approve it once you confirm that the separate test was needed @henrymartin1

Reply:
Don't worry about the error, happens to all of us, that's why the review is so important. And even then we introduce bugs and errors ;-)
Regarding your commment, I don't really understand it. Do want to say that we don't really need the new tests (I mean they are there yet, so 🤷‍♂️) or that we should have more tests?